llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp

```diff
@@ -568,6 +568,18 @@ bool RISCVInstructionSelector::select(MachineInstr &MI) {
     return true;
   }
 
+  // FIXME: We create a IMPLICIT_DEF and a G_CONSTANT in preISelLower when
+  // we encounter a G_SPLAT_VECTOR. We cannot select the G_CONSTANT until after
+  // the MI is lowered, since renderVLOp needs to see the G_CONSTANT. It would
+  // be nice if the InstructionSelector selected these instructions without
+  // needing to call select on them explicitly.
+  if (Opc == RISCV::G_VMV_V_X_VL || Opc == RISCV::G_VFMV_V_F_VL) {
```

Comment:
This broken if we create G_VMV_V_X_VL prior to instruction selection which we will need to do for VP intrinsics. You have to know that you started with a SPLAT_VECTOR.

Reply:
Added.